audio/out/ao.c

```diff
@@ -506,6 +507,16 @@ struct ao_device_list *ao_hotplug_get_device_list(struct ao_hotplug *hp)
     MP_TARRAY_APPEND(list, list->devices, list->num_devices,
         (struct ao_device_desc){"auto", "Autoselect device"});
 
+    if (!hp->ao && playback_ao && playback_ao->driver->hotplug_init) {
+        struct ao *ao = ao_alloc(true, hp->global, hp->wakeup_cb, hp->wakeup_ctx,
+                                 (char *)playback_ao->driver->name);
+        if (playback_ao->driver->hotplug_init(ao) >= 0) {
+            hp->ao = ao;
+        } else {
+            talloc_free(ao);
+        }
+    }
+
```

Comment:
I'm a bit confused about the approach here. If the playback ao has hotplug_init and it succeeds, then why would we want to iterate through the other drivers? It feels like it would be clearer to get the devices and skip the for loop. Unless I'm missing something here.

Reply:
Because the property `audio-device-list` is expected to contain devices of all AOs.

Reply:
Ok. Makes sense.